packages/gatsby-transformer-csv/src/gatsby-node.js

```diff
@@ -36,7 +36,7 @@ async function onCreateNode({ node, actions, loadNodeContent }, options) {
 
       return {
         ...obj,
-        id: obj.id ? obj.id : `${node.id} [${i}] >>> CSV`,
+        id: createNodeId(obj.id ? obj.id : `${node.id} [${i}] >>> CSV`),
```

Comment:
We should let people explicitly set IDs. There are use cases for that e.g. when creating mappings between nodes like we do for blog authors on gatsbyjs.org https://github.com/gatsbyjs/gatsby/blob/8261ac36f16915d06bd7d79a568bc0f77118dca2/www/gatsby-config.js#L7

In nodes created from potentially manually created data, we should leave IDs alone.